src/protocolhandler.ts

```diff
@@ -196,21 +191,20 @@ export class ProtocolHandler extends EventEmitter {
         if (this.closed) {
             return Promise.reject(NatsError.errorForCode(ErrorCode.CONN_CLOSED));
         }
+
         if (this.draining) {
             return Promise.reject(NatsError.errorForCode(ErrorCode.CONN_DRAINING));
         }
         this.draining = true;
 
-        let subs = this.subscriptions.all();
-        let promises: Promise<SubEvent>[] = [];
-        subs.forEach((sub) => {
-            let p = this.drainSubscription(sub.sid);
-            promises.push(p);
-        });
+        const subs = this.subscriptions.all();
+        let promises: Promise<SubEvent>[] = subs.map((sub) => this.drainSubscription(sub.sid));
```

Comment:
please add back the initialization - now if for some reason the code changes and the map doesn't have that key, the iteration will fail with an NPE

Reply:
Could you elaborate on that? `this.subscriptions.all()` returns a `Sub[]`, not nullable, so `subs.map` [always exists and always returns an array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map). If `subs` is empty, it'll return an empty array, so `promises` is initialized in all cases.

`sub.sid` is not safeguarded in the previous version either, so maybe I'm missing something?